pkg/api/types.go

```diff
@@ -910,6 +910,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// JobSpec is the specification of a job.
+// As the internal representation of a job, it may have either a TemplateRef or a Template set.
+type JobSpec struct {
+	// Completions is the number of desired successful completions of this job.
+	Completions int `json:"completions"`
+
+	// Selector is a label query over pods that should match the completed count.
+	Selector map[string]string `json:"selector"`
```

Comment:
I can see it work both ways, the simpler version is if the kubelet doesn't have a chance to report the completed status of the job, it could run multiple times. There was some discussion on https://github.com/GoogleCloudPlatform/kubernetes/issues/1624 about this case. The more involved fix would be to stash the expected completions locally, update it everytime a pod status matching the selectors changes to completed, and have the job controller monitor something like observedCompleted + active < spec.completions.

I think I'm going for the simpler version first, thoughts?